### DISCUSSION TITLE
Can inadyn be set to force an update as the service/daemon starts?

### DISCUSSION THREAD

**Discussion by ryanjrp** [troglobit/inadyn#460] [Q&A]
I have been noticing my dynamic DNS service is not always being updated by inadyn.  I suspect it is likely because the local inadyn cache matches the IP address being discovered as my system boots back up from an outage or maintenance.   Sometimes Dynu does not get information from inadyn to indicate the service has been restored.

I was wondering if it is possible without having to setup a cronjob to run once at boot with the inadyn -1 --force flags.

I use FreeBSD 13 with inadyn 2.11.0.  I have it setup as a service in the rc.conf file.  I was hoping to find a way in the inadyn.conf file to do a forced-update as the system launches the service during boot.  Currently I have decided to have the forced-update = 3600 which means my dynamic DNS service will be given the IP once an hour regardless of if anything changes.  This results in me having at worst a 1 hour additional outage for my site, should Dynu's DNS failover service not restore my DNS settings after a detected outage and my system coming backup.  I think there is something weird happening on the Dynu side of things. 

I spent time troubleshooting, by taking my system down, waiting for the failover, then bringing the system back up and checking Dynu for IP Update History on my account.  I found that my system was not reporting the IP being updated at the time of my system boot, so I started looking into if it was possible to force inadyn to do a forced update as the service/daemon started and cannot find a way to force it.

I am considering a cron job that only runs on system startup, and having the forced update to run at a longer interval like daily, weekly or even monthly as the default is.  Perhaps this could be considered a feature request?

Regards,

Ryan P

**Comment by troglobit** [comment]
You can enable full debugging with `-l debug` to ensure your theory first holds; "I suspect it is likely because the local inadyn cache matches the IP address being discovered ...".  You can also run inadyn in the foreground to see those logs in the terminal.

The whole point of this caching mechanism and preventing too frequent updates is that DDNS providers are known to lock accounts for "excessive use", and "unnecessary updates".

I'm not really sure what the problem is with your setup, if you have some clever interface fail-over mechanism, but if you can detect such a change you could clear the cache at first boot or fail-over.  This is effectively the same as the `--force` option, but if your account gets locked you know why.

Other cases are of course possible, as you describe, and I do recommend using other tools like cron to help you with that.

**Comment by cross** [comment]
Late followup on an old discussion.  @ryanjrp , I'm trying to get inadyn to work with Dynu, and it seems to me that dynu doesn't recognize the approach inadyn does/can use with it.  Let me know if you're still using inadyn with Dynu with any success.

**Reply by Hobbybyte** [reply]
Dynu Config :
https://github.com/troglobit/inadyn/issues/489#issuecomment-2633020997
